test/evp_test.c

```diff
@@ -2210,33 +2210,25 @@ static int encapsulate(EVP_TEST *t, EVP_PKEY_CTX *ctx, const char *op,
                        unsigned char **outwrapped, size_t *outwrappedlen,
                        unsigned char **outsecret, size_t *outsecretlen)
 {
-    int ret = 1;
+    int ret = 0;
     KEM_DATA *kdata = t->data;
     unsigned char *wrapped = NULL, *secret = NULL;
     size_t wrappedlen = 0, secretlen = 0;
-    OSSL_PARAM *params = NULL;
-    OSSL_PARAM_BLD *bld = NULL;
-    size_t params_n = 0, params_n_allocated = 0;
+    OSSL_PARAM params[10] = { OSSL_PARAM_END };
+    size_t params_n = 0;
 
-    if (sk_OPENSSL_STRING_num(kdata->init_ctrls) > 0) {
-        if (!TEST_ptr(params = OPENSSL_malloc(sizeof(OSSL_PARAM) * 2)))
-            goto err;
-        params[0] = OSSL_PARAM_construct_end();
-        params[1] = OSSL_PARAM_construct_end();
-        if (ctrl2params(t, kdata->init_ctrls, NULL, params, 2, &params_n))
+    if (sk_OPENSSL_STRING_num(kdata->init_ctrls) > 0)
+        if (ctrl2params(t, kdata->init_ctrls, NULL, params,
+                        OSSL_NELEM(params) - 1, &params_n))
             goto err;
-    }
 
-    if (kdata->entropy != NULL
-        && (!TEST_ptr(bld = OSSL_PARAM_BLD_new())
-            || !TEST_int_eq(OSSL_PARAM_BLD_push_octet_string(bld,
-                                                             OSSL_KEM_PARAM_MLKEM_ENC_ENTROPY,
-                                                             kdata->entropy,
-                                                             kdata->entropylen),
-                            1)
-            || !TEST_ptr(params = OSSL_PARAM_BLD_to_param(bld)))) {
-        ret = 0;
-        goto err;
+    if (kdata->entropy != NULL) {
```

Comment:
Why is this even needed?

Can't you just do this in the text file?
 #ikme= Entropy 
CtrlInit.ikme = hexikme:0303030303030303

The whole point of the ctrls is to avoid having to hardwire values in like this..